src/offchainapi/payment_logic.py

```diff
@@ -690,9 +721,31 @@ async def payment_process_async(self, payment, ctx=None):
             if current_status == Status.none:
                 await business.check_account_existence(new_payment, ctx)
 
+            # Provide KYC -- this may be async in case
+            # of need for user input
+            kyc_to_provide = await business.next_kyc_to_provide(
+                new_payment, ctx)
+
+            myself_new_actor = new_payment.data[role]
+
+            if Status.needs_kyc_data in kyc_to_provide:
+                extended_kyc = await business.get_extended_kyc(new_payment, ctx)
+                myself_new_actor.add_kyc_data(extended_kyc)
+
+            if Status.soft_match in kyc_to_provide:
+                additional_kyc = await business.get_additional_kyc(new_payment, ctx)
+                myself_new_actor.add_additional_kyc_data(additional_kyc)
+
+            if Status.needs_recipient_signature in kyc_to_provide:
+                signature = await business.get_recipient_signature(
+                    new_payment, ctx)
+                new_payment.add_recipient_signature(signature)
+
+            # Request more KYC Data or progress the protocol
```

Comment:
I think I did that because the request for KYC potentially changes the `current_status`, so I thought it was cleaner to come after the addition of KYC / Info that does not add status. Maybe this is only better in my mind.

Reply:
not sure I follow: var `current_status` is not updated between line 735-755?  :D

Reply:
Yeah, it is probably just in my mind.

Reply:
I want to reopen this discussion: now I feel the original order is better because: 
processing data from other side (`next_kyc_level_to_request`) first lets the acting side has more enough to make progresses with the payment. And the output of this can be used to determine what the acting side wants (`next_kyc_to_provide`). 
For example, in Novi, in `next_kyc_level_to_request`, we do risk assessment, and it may turn out to be a failure (resulting in abort) or a "needs_more_kyc_data" . And then in "next_kyc_to_provide", we provide signature when and only when the other side's data is good (which is checked in `next_kyc_level_to_request` called earlier). In the new order, we call `next_kyc_to_provide` and then `next_kyc_level_to_request`, we lose the natural ability to do so.

For your thoughts:
> I think I did that because the request for KYC potentially changes the current_status, so I thought it was cleaner to come after the addition of KYC / Info that does not add status. Maybe this is only better in my mind.

I don't really get it, so would like to hear more about the trade-offs

Reply:
resolved off line and PR is https://github.com/libra/off-chain-reference/pull/110